### DISCUSSION TITLE
Are Interleaved Vertex Buffers Preferred?

### DISCUSSION THREAD

**Discussion by joeedh** [gpuweb/gpuweb#2676] [Q&A] [answered]
I noticed a lot of examples are using interleaved vertex buffers.  Are they preferred over separate buffers? Thanks.

**Comment by Kangz** [comment, marked as answer]
In general interleaved vertex buffers are a bit more efficient than using separate vertex buffers because the memory for a single vertex is contiguous and that helps make caching behavior more efficient. But if your data comes from different buffers, there's no issue with using separate vertex buffers.
